jonason/JSONValue.cpp

```diff
@@ -0,0 +1,121 @@
+//
+//  JSONValue.cpp
+//  jonason
+//
+//  Created by Gabriel Lumbi on 2023-03-03.
+//
+
+#include "JSONValue.hpp"
+
+#include <utility>
+
+namespace jonason {
+
+JSONValue::JSONValue(decltype(OBJECT) tag): tag(tag) {
+    switch (tag) {
+        case OBJECT: new(&object) ObjectType(); break;
+        case ARRAY: new(&array) ArrayType(); break;
+        case STRING: new(&string) StringType(); break;
+        case NUMBER: number = 0; break;
+        case BOOLEAN: boolean = false; break;
+        case JSON_NULL: boolean = false; break;
+    }
+};
+
+JSONValue::JSONValue(const JSONValue& other)
+{
+    if (this == &other) { return; }
```

Comment:
I don't think this can happen since `this` doesn't exist before entering the constructor? However it can happen in your `operator=` implementation and it's missing there.